src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -1696,7 +1713,10 @@ type AdaptiveFSharpLspServer(workspaceLoader: IWorkspaceLoader, lspClient: FShar
           percentage = percentage 0 checksToPerform.Length
         )
 
-      do! Async.Parallel(checksToPerform, 2) |> Async.Ignore<unit array>
+      let maxConcurrency =
+        Math.Max(1.0, Math.Floor((float System.Environment.ProcessorCount) * 0.75))
```

Comment:
I did have this be conservative at the beginning. This tries to use 75% of cores. Maybe this should be a setting?